### ISSUE TITLE
Unexpected `The value of the associated %type_name% must be specified` error for `Fn*` types

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9730]
## Environment

* **IntelliJ Rust plugin version:** 0.4.183.4990-222-nightly
* **Rust toolchain version:** 1.67.0-nightly (85f4f41de 2022-11-08) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc.function-like, org.rust.macros.proc

## Problem description
After #5866, the plugin produces false positives for `Fn*` types that an associated type it not specified. But it's specified by returning type via special syntax for `Fn*` traits

<img width="793" alt="Screenshot 2022-11-14 at 15 20 50" src="https://user-images.githubusercontent.com/2539310/201683642-3a9700d9-3043-4872-bf9e-31700e06fa8a.png">

## Steps to reproduce
```rust
fn f(ff: &dyn Fn(i32) -> i32) {}
```

**Comment by Undin** [comment]
CC @dima74

**Comment by Kobzol** [comment]
Interesting, there's even a test for this.
I wanted to take a look, but I'm not able to compile the plugin at this moment though, because of this error:
```
Build file '/projects/personal/intellij-rust/build.gradle.kts' line: 51

An exception occurred applying plugin request [id: 'net.saliman.properties', version: '1.5.2']
> Failed to apply plugin 'net.saliman.properties'.
   > Could not create task ':setupDependencies'.
      > Failed to calculate the value of task ':setupDependencies' property 'idea'.
         > Failed to query the value of extension 'intellij' property 'version'.
            > The value for the 'intellij.version' property was not specified, see: https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#intellij-extension-version
```
Any hints? :)

**Comment by Undin** [comment]
@Kobzol I suppose some problem with Gradle IntelliJ plugin
I'm investigating it right now

**Comment by Undin** [comment]
@Kobzol error during building should be fixed now. If you still get the exception, call `gradle --refresh-dependencies`

**Comment by Undin** [comment]
About the initial issue - it seems it's because of changes in nightly stdlib. The same change leads to #9748. And it seems https://github.com/intellij-rust/intellij-rust/commit/630dc198e0ddbb9bff9be8da9a29a99ee997166d fixes it
